granian/server.py

```diff
@@ -466,13 +472,26 @@ def socket_loader():
             time.sleep(delay)
             logger.info(f'Stopping old worker-{idx + 1}')
             old_proc.terminate()
-            old_proc.join()
+            old_proc.join(self.workers_graceful_timeout)
+            # give it a short moment to exit
+            time.sleep(0.01)
```

Comment:
Without this change the server sometimes(1/10 times) went into the `if old_proc.proc.is_alive()` branch although the worker exited normally. I don't know why this happens. The sleep forces a "context switch"

Reply:
Wouldn't calling `kill` at that point still be ok (as it wouldn't do anything)?
10ms looks like _a lot of time_ to me.

Reply:
yes, it would be ok, but you get the false error in the log. 

(I am sure it would work with `time.sleep(0.001)`, too. Maybe even `time.sleep(0)`. I think it is about forcing a context switch.)

Reply:
+1 for using `os.sched_yield`

Reply:
tested `os.sched_yield`, unfortunately doesn't help.